### PR TITLE
repeated volume create information

### PR DESCRIPTION
**Pull request by zteBill**
**- when I created a volume with '-d' parameter to assign a driver**

**- I ran 'docker volume create -d bzk' then return error information: 
'Error response from daemon: 
create f8677134320e374273f0526b63279226448f2a681c3c33712f92d86622bd0841: 
create f8677134320e374273f0526b63279226448f2a681c3c33712f92d86622bd0841: 
Error looking up volume plugin bzk: 
legacy plugin: plugin not found' 
I think there is two create information is repeated and unnecessary**

**- I found the function which return the second 'create' information is 'CreateDriver()' then I modified 'create' to 'create driver'
I ran 'docker volume create -d bzk' then return error information:
'Error response from daemon: 
create 2094054c1a19c508267f2eeb2f97ff48b20214b40a501461fe0a614bd89b2f16: 
create driver 2094054c1a19c508267f2eeb2f97ff48b20214b40a501461fe0a614bd89b2f16: 
Error looking up volume plugin bzk: 
legacy plugin: plugin not found'
I think this is more clearer and more exacter then before 
otherWise I do not understand why 'legacy plugin' rather than 'illegal plugin', Because  there is not volume plugin bzk
**
**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: zteBill <bi.zhenkun@zte.com.cn>

### CODE REVIEW

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]
NOTLGTM, it is an error of the "create" operation.

Instead, I suggest modifying the `CreateWithRef()` function:
```go
func (s *VolumeStore) CreateWithRef(name, driverName, ref string, opts, labels map[string]string) (volume.Volume, error) {
        ...
	v, err := s.create(name, driverName, opts, labels)
	if err != nil {
		return nil, &OpErr{Err: err, Name: name, Op: "create"}
	}
        ...
}
```

If the type of the error returned from `create()` is `*OpErr`, you do not need to wrap it twice in another `*OpErr`.

**Comment by AkihiroSuda** [comment]
and nit about DCO: https://github.com/docker/docker/blob/master/CONTRIBUTING.md

> `Signed-off-by: Joe Smith <joe.smith@email.com>`
> Use your real name (sorry, no pseudonyms or anonymous contributions.)

**Comment by allencloud** [comment]
@zteBill  I am wondering if it is better to submit an issue to describe the problem more specifically. Then we can get more info about the background and version.

**Comment by zteBill** [comment]
but  the same error information returned from create() and CreateDriver() which  wraped twice
the information is "'Error response from daemon:
create f8677134320e374273f0526b63279226448f2a681c3c33712f92d86622bd0841:
create f8677134320e374273f0526b63279226448f2a681c3c33712f92d86622bd0841:"
I think it is inaccurate @allencloud

**Comment by zteBill** [comment]
As your suggest, I modify the 'create()' function,return 'err' object  directly when it is not null 
I run 'docker volume create -d bzk' then return error information:
"Error response from daemon: create driver c3a380ca16084007a7ea14aeb64ccc24d737ef4f05f81820968c6da7082a5437: Error looking up volume plugin bzk: legacy plugin: plugin not found" @AkihiroSuda

File: volume/store/store.go
Comment (AkihiroSuda) [inline volume/store/store.go]: Please revert this line

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

**Comment by zteBill** [comment]
I have reverted 'return nil, &OpErr{Op: "create driver", Name: name, Err: err}' to 'return nil, &OpErr{Op: "create", Name: name, Err: err}' @AkihiroSuda   Thanks~~

**Comment by zteBill** [comment]
It seems to have another problem that is unit test is failing
store_test.go
expected := &OpErr{Op: "create", Name: "fakeerror", Err: errors.New("create error")}
"if err != nil && err.Error() != expected.Error() {
		t.Fatalf("Expected create fakeError: create error, got %v", err)
	}"
I think the initial scheme which add 'create driver' information in createDriver() is better   Please give me your advice, Thanks  @AkihiroSuda

File: volume/store/store.go
Comment (zteBill) [inline volume/store/store.go]: I have reverted It seems to have another problem that is unit test is failing
store_test.go
expected := &OpErr{Op: "create", Name: "fakeerror", Err: errors.New("create error")}
"if err != nil && err.Error() != expected.Error() {
t.Fatalf("Expected create fakeError: create error, got %v", err)
}"
I think the initial scheme which add 'create driver' information in createDriver() is better Please give me your advice, Thanks

**Comment by AkihiroSuda** [comment]
Let's check the type of the error rather than just check whether the error is nil.
If the type of the error returned from `create()` is not `*OpErr`, you still need to wrap it in `*OpErr`.

**Comment by zteBill** [comment]
I check the type of error but result still wrap twice 
v, err := s.create(name, driverName, opts, labels)
	fmt.Println("err++", err.Error())
	checkErr := &OpErr{Op: "create", Name: name, Err: err}
	if err != nil && err.Error() != checkErr.Error(){
		return nil, &OpErr{Op: "create", Name: name, Err: err}
	} else {
		return nil, err
	}
I track these code, fmt print  "err++ create 3138e98c190e8475316a486a4b03743ea84f1c0f48067a3fe78f3f1d3dfc6a26: Error looking up volume plugin bzk: legacy plugin: plugin not found"
the error object is correct, but I do not know why it is not instance of *OpErr.
when I  wrap it in *OpErr, the result is still wrong
 "create 3138e98c190e8475316a486a4b03743ea84f1c0f48067a3fe78f3f1d3dfc6a26: 
create 3138e98c190e8475316a486a4b03743ea84f1c0f48067a3fe78f3f1d3dfc6a26: 
Error looking up volume plugin bzk: legacy plugin: plugin not found" 
I do not understand how it happened ? check type of error is not work
@AkihiroSuda

**Comment by AkihiroSuda** [comment]
I meant, please write a code that checks the type in store.go
I.e. `_, ok = err.(*OpErr)`

This check fails for the UT, because the error is from `fmt.Errorf()`
https://github.com/zteBill/docker/blob/412209d180e2fe14627395efa9c053ab38d4375b/volume/testutils/testutils.go

**Comment by zteBill** [comment]
why always I push code,  unit test check prompt: 

> Variable with name 'BUILD_DISPLAY_NAME' already exists, current value: '#28690', new value: '#28690'

> Error response from daemon: No such container: docker-pr35875

I run the follow commends each time:

> git add file 

> git fetch upstream master 

> git commit -s -m "description"

> git rebase -i upstream/master 

> git push -f  origin volume-errorinfo-inexact

Thanks @GordonTheTurtle

**Comment by AkihiroSuda** [comment]
```
08:28:48 These files are not properly gofmt'd:
08:28:48  - volume/store/store.go
08:28:48 
08:28:48 Please reformat the above files using "gofmt -s -w" and commit the result.
08:28:48 
```

**Comment by zteBill** [comment]
I have found this issue  I have ran 'gofmt -s -w ' then pushed  again. Thanks @AkihiroSuda

**Comment by zteBill** [comment]
4 checks have passed, what can I do next? Thanks @AkihiroSuda

**Review by AkihiroSuda — APPROVED** [review]
LGTM, thank you for working on this.
This PR will be merged if you can also get LGTM from another maintainer.

**Comment by AkihiroSuda** [comment]
@cpuguy83 PTAL?

**Comment by zteBill** [comment]
Thanks for your suggestion and help, thank you very much ! ~~ I am pleasant to participate docker project~ I would appreciate your guidance. @AkihiroSuda

**Comment by AkihiroSuda** [comment]
uh, and DCO should be your real name: https://github.com/docker/docker/blob/master/CONTRIBUTING.md

> `Signed-off-by: Joe Smith <joe.smith@email.com>`
> Use your real name (sorry, no pseudonyms or anonymous contributions.)

**Comment by zteBill** [comment]
OK, I will change it  Thanks @AkihiroSuda

**Comment by zteBill** [comment]
This PR has any other problems ?  Thanks @cpuguy83

**Comment by vdemeester** [comment]
ping @cpuguy83

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 🐻

**Pull request merged**
